kore/test/Test/Kore/Exec.hs

```diff
@@ -356,6 +362,105 @@ test_checkFunctions =
                 { sentenceAxiomAttributes = Attributes []
                 }
 
+test_checkBothMatch :: TestTree
+test_checkBothMatch =
+    testGroup
+        "checkBothMatch"
+        [ testCase "Function patterns do not both match." $ do
+            let verifiedModule =
+                    verifiedMyModule
+                        Module
+                            { moduleName = ModuleName "MY-MODULE"
+                            , moduleSentences =
+                                [ asSentence mySortDecl
+                                , asSentence $ constructorDecl "a"
+                                , --                                , asSentence $ constructorDecl "b"
+                                  asSentence mySymbDecl
+                                , mkEq "a" makeTruePredicate
+                                , mkEq "a" makeFalsePredicate
```

Comment:
I think this `mkEq` function is misleading, what do its arguments represent?

Reply:
This function takes arguments `name` and `pred` and produces the equation `f(x) = name, assuming pred`.  I think the name can be improved but I'm not sure what a good name for it would be.